packages/jest-jasmine2/src/jasmine/Suite.js

```diff
@@ -181,6 +181,33 @@ Suite.prototype.addExpectationResult = function() {
   }
 };
 
+function convertDescriptorToString(descriptor) {
+  if (
+    typeof descriptor === 'string' ||
+    typeof descriptor === 'number' ||
+    descriptor === undefined
+  ) {
+    return descriptor;
+  }
+
+  if (typeof descriptor !== 'function') {
+    throw new Error('describe expects a class, function, number, or string.');
+  }
+
+  if (descriptor.name !== undefined) {
+    return descriptor.name;
+  }
+
+  const stringified = descriptor.toString();
+  const typeDescriptorMatch = stringified.match(/class|function/);
+  const indexOfNameSpace =
+    typeDescriptorMatch.index + typeDescriptorMatch[0].length;
+  const indexOfNameAfterSpace = stringified.search(/\(|\{/, indexOfNameSpace);
```

Comment:
@JoshuaKGoldberg I'm looking at converting this to TS, and it's not happy about the second argument here. It's also not documented at MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/search

Should I just remove it?

Reply:
@SimenB ha, sure.

Looks like yet another "bug" fixed by TypeScript! 😍